docs/advanced.md

```diff
@@ -0,0 +1,156 @@
+The particular instrumentation types to apply can be specified using the `--trace` flag, which accepts the following values:
+
+* `cov`: AFL-style edge coverage
+* `cmp`: compares (int, long, String) and switch cases
+* `div`: divisors in integer divisions
+* `gep`: constant array indexes
+* `indir`: call through `Method#invoke`
+* `all`: shorthand to apply all available instrumentations (except `gep`)
+
+Multiple instrumentation types can be combined with a colon (Linux, macOS) or a semicolon (Windows).
+
+### Value profile
+
+The run-time flag `-use_value_profile=1` enables [libFuzzer's value profiling mode](https://llvm.org/docs/LibFuzzer.html#value-profile).
+When running with this flag, the feedback about compares and constants received from Jazzer's trace instrumentation is associated with the particular bytecode location and used to provide additional coverage instrumentation.
+See [ExampleValueProfileFuzzer.java](examples/src/main/java/com/example/ExampleValueProfileFuzzer.java) for a fuzz target that would be very hard to fuzz without value profile.
+
+### Custom hooks
+
+In order to obtain information about data passed into functions such as `String.equals` or `String.startsWith`, Jazzer hooks invocations to these methods.
+This functionality is also available to fuzz targets, where it can be used to implement custom sanitizers or stub out methods that block the fuzzer from progressing (e.g. checksum verifications or random number generation).
+See [ExampleFuzzerHooks.java](examples/src/main/java/com/example/ExampleFuzzerHooks.java) for an example of such a hook.
+An example for a sanitizer can be found in [ExamplePathTraversalFuzzerHooks.java](examples/src/main/java/com/example/ExamplePathTraversalFuzzerHooks.java).
+
+Method hooks can be declared using the `@MethodHook` annotation defined in the `com.code_intelligence.jazzer.api` package, which is contained in `jazzer_api_deploy.jar` (binary release) or built by the target `//agent:jazzer_api_deploy.jar` (Bazel).
+It is also available from [Maven Central](https://search.maven.org/search?q=g:com.code-intelligence%20a:jazzer-api).
+See the [javadocs of the `@MethodHook` API](https://codeintelligencetesting.github.io/jazzer-api/com/code_intelligence/jazzer/api/MethodHook.html) for more details.
+
+To use the compiled method hooks, they have to be available on the classpath provided by `--cp` and can then be loaded by providing the flag `--custom_hooks`, which takes a colon-separated list of names of classes to load hooks from.
+Hooks have to be loaded from separate JAR files so that Jazzer can [add it to the bootstrap class loader search](https://docs.oracle.com/javase/8/docs/api/java/lang/instrument/Instrumentation.html#appendToBootstrapClassLoaderSearch-java.util.jar.JarFile-).
+The list of custom hooks can alternatively be specified via the `Jazzer-Hook-Classes` attribute in the fuzz target JAR's manifest.
+
```

Comment:
Maybe also add a short section about setting no hooks (`-Djazzer.hooks=false` as a jvmArg) at all, like we use it in cifuzz coverage runs?

Some functionality / use cases (like ours) might need hooks disabled.
Not sure if this needs explanation here.

Reply:
I would rather not document it since it does more than just disable custom hooks (I know, the name isn't good): It disables any kind of instrumentation, including coverage. It is documented in `jazzer --help` and given how only wrappers such as `cifuzz` should need it, I would rather keep it out of the README unless users have a clear need for knowing about it.